src/mongocxx/test/spec/unified_tests/operations.cpp

```diff
@@ -0,0 +1,1007 @@
+
+    auto indexes = coll.indexes().list();
+    auto expected_name = op["arguments"]["indexName"].get_string().value.to_string();
+
+    return std::any_of(indexes.begin(), indexes.end(), [&](document::view index) {
+        return expected_name == index["name"].get_string().value.to_string();
+    });
+}
+
+struct with_transaction_cb {
+    array::value operations;
+    entity::map& map;
+    spec::apm_checker& apm;
+
+    void operator()(client_session*) {
+        for (auto&& op : operations.view()) {
+            operations::run(map, apm, op);
+        }
+    }
+};
+
+document::value download(gridfs::bucket& bucket, document::view op) {
+    using bsoncxx::types::bson_value::value;
+
+    auto id = op["arguments"]["id"].get_value();
+
+    std::ostringstream result{};
+    bucket.download_to_stream(id, &result);
+
+    auto result_str = result.str();
+    std::vector<std::uint8_t> bytes{result_str.begin(), result_str.end()};
+
+    return make_document(kvp("result", value(bytes)));
+}
+
+document::value upload(entity::map& map, document::view op) {
+    auto key = op["saveResultAsEntity"].get_string().value.to_string();
+    auto arguments = op["arguments"].get_document().value;
+    auto object = op["object"].get_string().value.to_string();
+    auto& bucket = map.get_bucket(object);
+
+    options::gridfs::upload upload_options;
+
+    if (arguments["chunkSizeBytes"]) {
+        upload_options.chunk_size_bytes(arguments["chunkSizeBytes"].get_int32().value);
+    }
+
+    if (arguments["metadata"]) {
+        upload_options.metadata(arguments["metadata"].get_document().value);
+    }
+
+    auto filename = arguments["filename"].get_string().value;
+    auto uploader = bucket.open_upload_stream(filename, upload_options);
+    auto source = arguments["source"].get_document().value;
+    auto hex = source["$$hexBytes"].get_string().value;
+    auto source_bytes = test_util::convert_hex_string_to_bytes(hex);
+
+    uploader.write(source_bytes.data(), source_bytes.size());
+    auto upload_result = uploader.close();
+    auto id = upload_result.id();
+
+    map.insert(key, id);
+    return make_document(kvp("result", id));
+}
+
+document::value delete_one(collection& coll, document::view operation) {
+    document::view arguments = operation["arguments"].get_document().value;
+    document::view filter = arguments["filter"].get_document().value;
+    options::delete_options options{};
+
+    if (arguments["collation"]) {
+        options.collation(arguments["collation"].get_document().value);
+    }
+
+    if (arguments["hint"]) {
+        if (arguments["hint"].type() == bsoncxx::v_noabi::type::k_string)
+            options.hint(hint{arguments["hint"].get_string().value});
+        else
+            options.hint(hint{arguments["hint"].get_document().value});
+    }
+
+    auto result = builder::basic::document{};
+    std::int32_t deleted_count = 0;
+
+    if (auto delete_result = coll.delete_one(filter, options)) {
```

Comment:
This should check if a session was passed.

Reply:
Done.